### PR TITLE
libhugetlbfs handling with header flag: RHEL9 

### PR DESCRIPTION
**Pull request by sasingh87**
Since RHEL9 onwards support for libhugetlbfs not there, set a flag HAVE_HUGETLB_HEADER to handle at appropriate places.
Aggregated from  PR -> Add rhel-9 distro check to cancel the test #2435
Signed-off-by: Saurabh Singh <Saurabh.Singh2@amd.com>

### CODE REVIEW

File: memory/migrate_pages.py [hunk 1]
```diff
@@ -50,16 +50,23 @@ def setUp(self):
...
             pkgs.extend(['libpthread-stubs0-dev',
                          'libnuma-dev', 'libhugetlbfs-dev'])
-        elif dist.name in ["centos", "rhel", "fedora"]:
-            pkgs.extend(['numactl-devel', 'libhugetlbfs-devel'])
-        elif dist.name == "SuSE":
+        elif self.dist.name in ["centos", "rhel", "fedora"]:
+            if (self.dist.name == 'rhel' and self.dist.version == '9'):
+                self.log.info("libhugetlbfs is not available RHEL 9.x onwards,\
+                                so tests related to hugepage will be cancelled")
+                pkgs.extend(['numactl-devel'])
```
Comment (sacsant) [inline memory/migrate_pages.py:62]: Move this outside of RHEL 9 case.

File: memory/migrate_pages.py [hunk 2]
```diff
@@ -50,16 +50,23 @@ def setUp(self):
...
+            else:
+                pkgs.extend(['numactl-devel', 'libhugetlbfs-devel'])
```
Comment (sacsant) [inline memory/migrate_pages.py:64]: Adjust this accordingly

File: memory/migrate_pages.py.data/node_move_pages.c [hunk 3]
```diff
@@ -327,21 +329,31 @@ int main(int argc, char *argv[])
...
-                case 'o':
-			overcommit = 1;
-                        break;
                 case 't':
 			thp = 1;
                         break;
-                case 'h':
+#ifdef HAVE_HUGETLB_HEADER
+               case 'o':
+
```
Comment (sacsant) [inline memory/migrate_pages.py.data/node_move_pages.c:337]: Remove the blank line

File: memory/migrate_pages.py.data/node_move_pages.c [hunk 4]
```diff
@@ -327,21 +329,31 @@ int main(int argc, char *argv[])
...
+                         overcommit = 1;
+                         break;
+
```
Comment (sacsant) [inline memory/migrate_pages.py.data/node_move_pages.c:340]: same here

File: memory/migrate_pages.py.data/node_move_pages.c [hunk 5]
```diff
@@ -327,21 +329,31 @@ int main(int argc, char *argv[])
...
+
+		case 'h':
+
```
Comment (sacsant) [inline memory/migrate_pages.py.data/node_move_pages.c:343]: and here

File: memory/migrate_pages.py [hunk 6]
```diff
@@ -50,16 +50,23 @@ def setUp(self):
...
+                exp_cmd = "export HAVE_HUGETLB_HEADER"
+                process.system(exp_cmd, shell=True, sudo=True, ignore_status=True)
```
Comment (sacsant) [inline memory/migrate_pages.py:66]: Introduces a code style warning for >80 characters. Split this statement accordingly.

File: memory/migrate_pages.py [hunk 7]
```diff
@@ -86,9 +93,14 @@ def test(self):
         cmd = './node_move_pages -n %s' % self.nr_chunks
 
         if self.hpage:
-            cmd += ' -h'
-            if self.hpage_commit:
-                cmd += ' -o'
+           if (self.dist.name == 'rhel' and self.dist.version == '9'):
+               if self.hpage_commit:
+                  self.cancel("Hugepage(over commit) tests are cancelled on RHEL-9")
```
Comment (sacsant) [inline memory/migrate_pages.py:98]: Introduces a code style warning for >80 characters. The if hpage_commit block isn't necessarily required. Cancel the test if hpage is set scenario will also cover hpage_commit case as well.

**Review by sacsant — CHANGES_REQUESTED** [review]

File: memory/migrate_pages.py (hunk 1 above)
Comment (sasingh87) [reply memory/migrate_pages.py:62]: Sure, so we would only have  "numactl-devel" package for all centos, fedora and rhel-9 , I  believe !

File: memory/migrate_pages.py (hunk 2 above)
Comment (sasingh87) [reply memory/migrate_pages.py:64]: 58         elif self.dist.name in ["centos", "rhel", "fedora"]:
 59             if (self.dist.name == 'rhel' and self.dist.version == '9'):
 60                 self.log.info("hugepage tests would be cancelled due to \
 61                                 non-availability of libhugetlbfs in RHEL-9")
 62             else:
 63                 #pkgs.extend(['numactl-devel', 'libhugetlbfs-devel'])
 64                 exp_cmd = "export HAVE_HUGETLB_HEADER"
 65                 process.system(exp_cmd, shell=True, sudo=True,\
 66                         ignore_status=True)
 67             pkgs.extend(['numactl-devel'])
**I assume these changes should be fine !**

File: memory/migrate_pages.py (hunk 1 above)
Comment (sacsant) [reply memory/migrate_pages.py:62]: > Sure, so we would only have "numactl-devel" package for all the three, centos, fedora and rhel-9 , I believe !

No. numactl for all centos, fedora and RHEL (including 9)
libhugetlbfs for all centos, fedora and RHEL (excluding 9)
 
if (RHEL, Fedora, CentOS)
     packages.extend(numactl-devel)
     if (!rhel9)
           packages.extend(libhugetlbfs-devel)

**Comment by sacsant** [comment]
@sasingh87 Can you merge all these changes into a single commit?

**Comment by sacsant** [comment]
commit [2a8837d](https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2447/commits/2a8837de882621250feb23d1993c78c568636209) seems to have pulled in a lot of extra changes unrelated to this PR. You should not create separate commits to incorporate review changes (unless specifically asked to do)

Here’s what you should do

1. revert all the changes (submitted originally via this PR) from your local copy (use the same branch that you had used to raise the PR)
2. Redo the original changes + any additional changes to incorporate review comments.
3. commit the changes into a single patch
4. force push the local branch to GitHub.

**Comment by PraveenPenguin** [comment]
@sasingh87 can you please check @sacsant request as squash commits in valid commit

**Comment by kalpanashtty** [comment]
@PraveenPenguin  can you please delete this PR(PR submitter left the company). 
I have resent this PR, please review it - https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2513

**Comment by PraveenPenguin** [comment]
@kalpanashtty sure thanks, will close this PR and we will pursue discussion on #2513

**Pull request closed without merging**